app/models/user.rb

```diff
@@ -0,0 +1,18 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+  validates :name, presence: true,
+  length: {maximum: Settings.validate.names.max_length}
+  validates :email, presence: true,
+  length: {maximum: Settings.validate.email.max_length},
+  format: {with: Settings.regex.email},
+  uniqueness: true
+  has_secure_password
+  validates :password, presence: true,
+   length: {minimum: Settings.validate.password.min_length}
+
+  private
+
+  def downcase_email
+    self.email.downcase!
```

Comment:
em đã test thử trường hợp ko nhập email chữ thường chưa em 
```suggestion
    email.downcase!
```

Reply:
Dạ có test rồi, thuộc tính email không phân biệt hoa thường, có thể dùng thuộc tính case_sensitive: false của validate uniqueness, nhưng để tối ưu chỉ mục index cho email thì mình sẽ ép dữ liệu về chữ thường trước khi lưu vào database.